app/src/main/java/com/example/makeiteven2/adapters/LevelsAdapter.kt

```diff
@@ -0,0 +1,84 @@
+package com.example.makeiteven2.adapters
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+import androidx.recyclerview.widget.RecyclerView
+import com.example.makeiteven2.R
+import com.example.makeiteven2.data_models.Level
+import com.example.makeiteven2.extras.OnTouchAnimation
+import kotlinx.android.synthetic.main.level_cell.view.*
+
+class LevelsAdapter(private var mLevelsList : List<Level> = ArrayList(), private val context: Context,private var mCurrentStage : Int)
+    : RecyclerView.Adapter<LevelsAdapter.ViewHolder>(){
+
+    class ViewHolder(private val view : View,private val context: Context) : RecyclerView.ViewHolder(view){
+        private val onTouchAnimation = OnTouchAnimation(context)
+        private lateinit var levelBtn : Button
```

Comment:
private val onTouchAnimation = OnTouchAnimation(context)
        private lateinit var levelBtn : Button
        private lateinit var lockStage : View
        private lateinit var stageNumberLayout : View
        private lateinit var stageNumberTv : TextView

don't have to declare it here, just use them inside if bind function, redundant

Reply:
changed it a bit